cpp/ql/src/Likely Bugs/Memory Management/ReturnStackAllocatedMemory.ql

```diff
@@ -27,16 +27,26 @@ class ReturnStackAllocatedMemoryConfig extends MustFlowConfiguration {
   ReturnStackAllocatedMemoryConfig() { this = "ReturnStackAllocatedMemoryConfig" }
 
   override predicate isSource(Instruction source) {
-    // Holds if `source` is a node that represents the use of a stack variable
-    exists(VariableAddressInstruction var, Function func |
-      var = source and
-      func = source.getEnclosingFunction() and
-      var.getAstVariable() instanceof StackVariable and
-      // Pointer-to-member types aren't properly handled in the dbscheme.
-      not var.getResultType() instanceof PointerToMemberType and
+    exists(Function func |
       // Rule out FPs caused by extraction errors.
       not any(ErrorExpr e).getEnclosingFunction() = func and
-      not intentionallyReturnsStackPointer(func)
+      not intentionallyReturnsStackPointer(func) and
+      func = source.getEnclosingFunction()
+    |
+      // `source` is an instruction that represents the use of a stack variable
+      exists(VariableAddressInstruction var |
+        var = source and
+        var.getAstVariable() instanceof StackVariable and
+        // Pointer-to-member types aren't properly handled in the dbscheme.
+        not var.getResultType() instanceof PointerToMemberType
+      )
+      or
+      // `source` is an instruction that represents the return value of a
+      // function that is known to return stack-allocated memory.
+      exists(Call call |
+        call.getTarget().hasGlobalName(["alloca", "strdupa", "strndupa"]) and
```

Comment:
mscv has `_alloca`. Maybe also add that.

Reply:
Good call. I also see that there's https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/malloca?view=msvc-170. I will add these

Reply:
(I see that `_malloca ` could potentially be safe if it's guaranteed to allocate memory above the `_ALLOCA_S_THRESHOLD` threshold, but I don't think this should exclude it from the query.)

Reply:
Fixed in https://github.com/github/codeql/commit/bae7e10e463c7e62e80f60daa5b5508f3eddf076